executor/src/constant_evaluator/mod.rs

```diff
@@ -69,10 +70,19 @@ fn generate_values<T: FieldElement>(
             };
             let index_expr;
             let e = if let Some(index) = index {
-                index_expr = Expression::IndexAccess(IndexAccess {
-                    array: e.clone().into(),
-                    index: Box::new(Expression::Number(index.into(), None)),
-                });
+                index_expr = Expression::IndexAccess(
+                    SourceRef::unknown(),
+                    IndexAccess {
```

Comment:
I'm wondering if it  makes sense to implement Into from IndexAccess to Expression, where the source ref is set to unknown (and of course for all other structs inside expression). Could that lead to us not setting the source ref even though it is known?